geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/lifecycle/StartJConsoleCommandTest.java

```diff
@@ -12,29 +12,89 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
+
 package org.apache.geode.management.internal.cli.commands.lifecycle;
 
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
 
-import org.apache.geode.management.internal.cli.i18n.CliStrings;
+import org.apache.geode.management.internal.cli.shell.Gfsh;
+import org.apache.geode.test.junit.rules.GfshParserRule;
 
 public class StartJConsoleCommandTest {
```

Comment:
Curious, do we not have more end-to-end test coverage for this command?

Reply:
for the changes we are making, a junit test should do. An end to end test would need to fire up additional process, hard to control in a test environment. 

We should always prefer junit test over integration test and integration over dunit tests. Use dunit tests only when necessary.

Reply:
Oh, that statement should have a clause: with the same test coverage, junit is preferred than integration test, integration test is preferred than dunit

Reply:
Ok, that makes sense. Thanks for clarifying!